core/src/main/java/org/elasticsearch/index/query/MoreLikeThisQueryParser.java

```diff
@@ -95,7 +95,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         XContentParser parser = parseContext.parser();
 
         MoreLikeThisQuery mltQuery = new MoreLikeThisQuery();
-        mltQuery.setSimilarity(parseContext.searchSimilarity());
+        mltQuery.setSimilarity(parseContext.context().searchSimilarity());
```

Comment:
this should maybe be moved to the data node as well once we refactor this query?